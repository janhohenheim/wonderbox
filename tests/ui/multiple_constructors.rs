use wonderbox::autoresolvable;

struct List {
    capacity: usize,
}

#[autoresolvable]
impl List {
    fn new() -> Self {
        List { capacity: 0 }
    }

    fn with_capacity(capacity: usize) -> Self {
        List { capacity }
    }
}

fn main() {}
