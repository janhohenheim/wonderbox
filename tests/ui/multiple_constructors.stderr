error: Expected one constructor, found 2
 --> $DIR/multiple_constructors.rs:8:1
  |
8 | impl List {
  | ^^^^
